src/core/effects/DepthOfField.vue

```diff
@@ -64,54 +65,21 @@ const unwatch = watchEffect(() => {
   composer?.value?.addPass(pass.value)
 })
 
-watchEffect(() => {
-  if (!effect.value) return
-  const plainEffectPass = new DepthOfFieldEffect()
-
-  // blendFunction is not updated, because it has no setter in BloomEffect
-
-  effect.value.circleOfConfusionMaterial.worldFocusDistance
-    = props.worldFocusDistance !== undefined
-      ? props.worldFocusDistance
-      : plainEffectPass.circleOfConfusionMaterial.worldFocusDistance
-
-  effect.value.circleOfConfusionMaterial.worldFocusRange
-    = props.worldFocusRange !== undefined
-      ? props.worldFocusRange
-      : plainEffectPass.circleOfConfusionMaterial.worldFocusRange
-
-  effect.value.circleOfConfusionMaterial.focusRange
-    = props.focusRange !== undefined
-      ? props.focusRange
-      : plainEffectPass.circleOfConfusionMaterial.focusRange
-
-  effect.value.circleOfConfusionMaterial.focusDistance
-    = props.focusDistance !== undefined
-      ? props.focusDistance
-      : plainEffectPass.circleOfConfusionMaterial.focusDistance
-
-  effect.value.bokehScale
-    = props.bokehScale !== undefined
-      ? props.bokehScale
-      : plainEffectPass.bokehScale
-
-  effect.value.blurPass.resolution.scale
-    = props.resolutionScale !== undefined
-      ? props.resolutionScale
-      : plainEffectPass.blurPass.resolution.scale
-
-  effect.value.blurPass.resolution.width
-    = props.resolutionX !== undefined
-      ? props.resolutionX
-      : plainEffectPass.resolution.width
-
-  effect.value.blurPass.resolution.height
-    = props.resolutionX !== undefined
-      ? props.resolutionX
-      : plainEffectPass.resolution.height
-
-  plainEffectPass.dispose()
-})
+makePropWatchers(
```

Comment:
This looks amazing, please explain to me later, I would like to use in cientos too

Reply:
Thanks! Yes, I think this can come in handy in all of our packages. Just send me a message and I'll explain it to you.